lib/src/main/java/com/nextcloud/android/sso/SingleSignOn.java

```diff
@@ -0,0 +1,52 @@
+package com.nextcloud.android.sso;
+
+import com.nextcloud.android.sso.model.FilesAppType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+public class SingleSignOn {
+    private static final SingleSignOn SINGLE_SIGN_ON = new SingleSignOn();
+    private List<FilesAppType> types = new ArrayList<>();
+
+    public SingleSignOn() {
+        types.add(FilesAppType.PROD);
+        types.add(FilesAppType.QA);
+        types.add(FilesAppType.DEV);
+    }
+
+    public static SingleSignOn getSingleSignOn() {
+        return SINGLE_SIGN_ON;
+    }
+
+    public synchronized void init(FilesAppType type) {
+        types.clear();
+        types.add(type);
+    }
+
+    public List<FilesAppType> getTypes() {
+        return types;
+    }
+
+    public String[] getAccountTypes() {
+        return types.stream().map(a -> a.accountType).toArray(String[]::new);
+    }
+
+
+    /**
+     * @return {@link #FilesAppType.PROD}, {@link #QA} or {@link #DEV} depending on {@param accountType}.
+     * Uses {@link #PROD} as fallback.
+     */
+    @NonNull
+    public FilesAppType findByAccountType(@Nullable String accountType) {
+        for (final var type : types) {
+            if (type.accountType.equalsIgnoreCase(accountType)) {
+                return type;
+            }
+        }
+        return FilesAppType.PROD;
```

Comment:
Careful here, if `FilesAppType` is not included in the `types` list, throwing an exception might be better to prevent an invalid state, no? Also the Javadoc comment should be updated.